syntax = "proto3";

package backendservice;

service PortfolioService {
    rpc postCsv (PortfolioCSVRequest) returns (PortfolioCSVResponse);
    
    rpc putPortfolio (PortfolioCSVRequest) returns (PortfolioCSVResponse);

    rpc getPortfolioInfo (PortfolioInfoRequest) returns (PortfolioInfoResponse);

    rpc addRecomendedValues (RecomendedValuesRequest) returns (RecomendedValuesResponse);
}

message DataInfo{
    string stockSymbol = 1;
    string priceOfPurchase = 2;
    int32 count = 3;
    int64 priceCurrent = 4;
    EPrediction prediction = 5;
}

message PortfolioCSVRequest {
    int64 id = 1;  //client id
    string filename = 2;          // Имя файлаs
    bytes content = 3;            // Содержимое файла (в бинарном виде)
}

message PortfolioCSVResponse {
    bool success = 1;             // Успешно ли завершилась операция
    string message = 2;           // Сообщение с результатом
}


enum EInfoType {
    FULL = 0; //full info 
    RISED = 2; //only rised 
    FALLEN = 3; //only fallen
}

enum EPrediction {
    STAY = 0; //it is rising more
    SELL = 1; //time to sell 
    UNPROFITABLE = 2; //no reason to sell 
}

message PortfolioInfoRequest {
    int64 id = 1;//client id
    EInfoType infoType = 2;
}

message PortfolioInfoResponse {
    repeated DataInfo infos = 1;
}

message RecomendedValue {
    string name = 1;
    int32 count = 2;
    int64 price = 3;
}


message RecomendedValuesRequest{
    int64 id = 1; //client id
    repeated RecomendedValue values = 2;    
}

message RecomendedValuesResponse {
    string status = 1;
}
